src/CommonLib/Helpers.cs

```diff
@@ -24,6 +24,39 @@ public static class Helpers
             "S-1-5-19", "S-1-5-20", "S-1-0-0", "S-1-0", "S-1-2-1"
         };
 
+        public static string RemoveDistinguishedNamePrefix(string distinguishedName)
+        {
+            if (!distinguishedName.Contains(","))
+            {
+                return "";
+            }
+            if (distinguishedName.IndexOf("DC=", StringComparison.OrdinalIgnoreCase) < 0)
+            {
+                return "";
+            }
+
+            for (var i = distinguishedName.IndexOf(','); i > -1; i = distinguishedName.IndexOf(',', i + 1))
```

Comment:
I found this logic pretty hard to parse. It appears correct after looking through usage and tests but I think I'd want to see a comment at the very least explaining what this for loop is doing and why it's necessary